blog/views.py

```diff
@@ -4,105 +4,105 @@
+from django.views.decorators.http import require_POST
 
 
 def post_list(request):
-    posts = Post.objects.filter(published_date__lte=timezone.now()).order_by('published_date')
-    data = serializers.serialize('json', posts)
+    posts = Post.objects.filter(published_date__lte=timezone.now()).order_by(
+        "published_date"
+    )
+    data = json.dumps(
+        [model_to_dict(post) for post in posts], cls=CustomDateTimeJSONEncoder
+    )
 
-    return JsonResponse({'data': data}, status=HTTPStatus.OK)
+    return JsonResponse({"data": data}, status=HTTPStatus.OK)
 
 
 def post_detail(request, pk):
     post = get_object_or_404(Post, pk=pk)
-    data = serializers.serialize('json', [post, ])
 
-    return JsonResponse({'data': data}, status=HTTPStatus.OK)
+    return JsonResponse(model_to_dict(post), status=HTTPStatus.OK)
 
 
 @login_required
 def post_new(request):
     form = PostForm
-    return render(request, 'blog/post_edit.html', {'form': form})
+    return render(request, "blog/post_edit.html", {"form": form})
 
 
 @login_required
+@require_POST
 def post_new(request):
-    if request.method == 'POST':
-        form = PostForm(request.POST)
-        if form.is_valid():
-            post = form.save(commit=False)
-            post.author = request.user
-            post.save()
-            data = serializers.serialize('json', [post, ])
-            return JsonResponse({'data': data}, status=HTTPStatus.CREATED)
-        else:
-            message = {"message": "잘못된 입력입니다"}
-            return JsonResponse(message, status=HTTPStatus.BAD_REQUEST)
+    data = json.loads(request.body)
+
+    if "title" in data and "text" in data:
+        post = Post.objects.create(
+            author=request.user, title=data["title"], text=data["text"]
+        )
+        post.save()
+        return JsonResponse(model_to_dict(post), status=HTTPStatus.CREATED)
     else:
-        form = PostForm()
-    return render(request, 'blog/post_edit.html', {'form': form})
+        return JsonResponse({"message": "잘못된 입력입니다"}, status=HTTPStatus.BAD_REQUEST)
 
 
 @login_required
+@require_POST
 def post_edit(request, pk):
     post = get_object_or_404(Post, pk=pk)
-    if request.method == 'POST':
-        form = PostForm(request.POST, instance=post)
-        if form.is_valid():
-            post = form.save(commit=False)
-            post.author = request.user
-            post.save()
-            return redirect('post_detail', pk=post.pk)
+    data = json.loads(request.body)
+
+    if "title" in data and "text" in data:
+        post.title = data["title"]
+        post.text = data["text"]
+        post.author = request.user
+        post.save()
+        return JsonResponse(model_to_dict(post), status=HTTPStatus.OK)
     else:
-        form = PostForm(instance=post)
-    return render(request, 'blog/post_edit.html', {'form': form})
+        return JsonResponse({"message": "잘못된 입력입니다"}, status=HTTPStatus.BAD_REQUEST)
```

Comment:
https://suwoni-codelab.com/python%20%EA%B8%B0%EB%B3%B8/2018/03/06/Python-Basic-EAFP/
이건 EAFP 관련 글인데 적용하면 대충 이렇게 해볼 수 있을 것 같네요
```suggestion
    try:
        post.title = data["title"]
        post.text = data["text"]
    except KeyError:
        return JsonResponse({"message": "잘못된 입력입니다"}, status=HTTPStatus.BAD_REQUEST)
    else:
        post.author = request.user
        post.save()
        return JsonResponse(model_to_dict(post), status=HTTPStatus.OK)
```

Reply:
5beda40 적용했습니다!